version: '3'

services:
  portainer:
    image: portainer/portainer-ce:alpine
    container_name: portainer
    command: -H unix:///var/run/docker.sock
    expose:
      - "9000"
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "portainer_data:/data"
    restart: always
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.1

  db:
    image: postgres:latest
    container_name: my-postgres
    environment:
      POSTGRES_DB: postgres
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
    expose:
      - "5432"
    volumes:
      - db_data:/var/lib/postgresql/data
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.2

  nginx:
    image: nginx:latest
    container_name: nginx
    expose:
      - "80:80"
    build:
      context: ./nginx-proxy
    environment:
    - NGINX_HOST=foobar.com
    - NGINX_PORT=80
    volumes:
      - ./templates:/etc/nginx/templates
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.3

  filebrowser:
    image: filebrowser/filebrowser:latest
    container_name: filebrowser
    ports:
      - "8080:80"
    volumes:
      - filebrowser_data:/data
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.5

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4
    environment:
        PGADMIN_DEFAULT_PASSWORD: postgres
        PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
        - pgadmin:/var/lib/pgadmin
    ports:
        - "5432"
    networks:
        - revproxy-apps
    restart: unless-stopped

  geoserver:
    image: vulhub/geoserver:2.22.1
    volumes:
        - geoserver:/opt/geoserver/data_dir
    ports:
        - "8080"
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.4
    restart: on-failure

  jupyterlab:
    image: cropsinsilico/jupyterlab:latest
    container_name: jupyterlab
    ports:
      - "8888:8888"
    volumes:
      - jupyterlab_data:/home/jovyan/work
    environment:
      JUPYTER_ENABLE_LAB: "yes"
    networks:
      revproxy-apps:
        ipv4_address: 172.24.0.4

volumes:
  portainer_data:
  db_data:
  filebrowser_data:
  pgadmin:
  geoserver:
  jupyterlab_data:

networks:
  revproxy-apps:
    name: revproxy-apps
    external: true
    ipam:
      config:
        - subnet: "172.24.0.0/16"
  